Source/Artifacts/Artifacts.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Dolittle.SDK.Artifacts
+{
+    /// <summary>
+    /// Represents an implementation of <see cref="IArtifacts{TArtifact}" />.
+    /// </summary>
+    /// <typeparam name="TArtifact">The <see cref="Type" /> of <see cref="Artifact" />.</typeparam>
+    public abstract class Artifacts<TArtifact> : IArtifacts<TArtifact>
+        where TArtifact : Artifact
+    {
+        readonly IDictionary<TArtifact, Type> _artifactToTypeMap;
+        readonly IDictionary<Type, TArtifact> _typeToArtifactMap;
+        readonly ILogger<Artifacts<TArtifact>> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Artifacts{TArtifact}"/> class.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger" />.</param>
+        protected Artifacts(ILogger<Artifacts<TArtifact>> logger)
+        {
+            _logger = logger;
+            _artifactToTypeMap = new Dictionary<TArtifact, Type>();
+            _typeToArtifactMap = new Dictionary<Type, TArtifact>();
+        }
```

Comment:
```suggestion
    public abstract class Artifacts<TArtifact> : IArtifacts<TArtifact>
        where TArtifact : Artifact
    {
        readonly IDictionary<TArtifact, Type> _artifactToTypeMap;
        readonly IDictionary<Type, TArtifact> _typeToArtifactMap;
        readonly ILogger _logger;

        /// <summary>
        /// Initializes a new instance of the <see cref="Artifacts{TArtifact}"/> class.
        /// </summary>
        /// <param name="logger">The <see cref="ILogger" />.</param>
        protected Artifacts(ILogger logger)
        {
            _logger = logger;
            _artifactToTypeMap = new Dictionary<TArtifact, Type>();
            _typeToArtifactMap = new Dictionary<Type, TArtifact>();
        }
```
We don't need to specify the type of the given `ILogger`, the class doesn't really care what's the type of the logger, just that it's created for it. It is still given the correct typed logger from the factory.